Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,410 @@
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                GestureUpdated( eventData.selectedObject, eventData );
+            }
+        }
+
+        public void OnGestureCompleted( InputEventData eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                GestureCompleted( eventData.selectedObject, eventData );
+            }
+
+        }
+        public void OnGestureCompleted( InputEventData<float> eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                GestureCompleted( eventData.selectedObject, eventData );
+            }
+
+        }
+        public void OnGestureCompleted( InputEventData<Vector2> eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                GestureCompleted( eventData.selectedObject, eventData );
+            }
+
+        }
+
+        public void OnGestureCompleted( InputEventData<Vector3> eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                GestureCompleted( eventData.selectedObject, eventData );
+            }
+
+        }
+
+        public void OnGestureCompleted( InputEventData<Quaternion> eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                GestureCompleted( eventData.selectedObject, eventData );
+            }
+
+        }
+        
+        public void OnGestureCanceled( InputEventData eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                GestureCanceled( eventData.selectedObject, eventData );
+            }
+        }
+        public void OnInputUp( InputEventData eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                InputUp( eventData.selectedObject, eventData );
+            }
+        }
+
+        public void OnInputDown( InputEventData eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                InputDown( eventData.selectedObject, eventData );
+            }
+        }
+
+        public void OnInputPressed( InputEventData<float> eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                InputPressed( eventData.selectedObject, eventData );
+            }
+        }
+
+        public void OnPositionInputChanged( InputEventData<Vector2> eventData )
+        {
+            if ( IsInteractable( eventData.selectedObject ) )
+            {
+                PositionInputChanged( eventData.selectedObject, eventData );
+            }
+        }
+
+        #endregion
+
+        #region Protected Virtual Callback Functions
+
+        protected virtual void FocusEnter(GameObject obj, FocusEventData eventData ) { }
```

Comment:
Not sure how I feel about making "clone" methods of the handlers. What's the advantage here?

Reply:
This is how it worked previously, do you see a better approach? The important thing is the interactable list check. If the extending class overrides the original handler method, it will git every event call from `InputSystem` - where we really only want extending classes to get the event if interactable list check passes.

Reply:
I don't see why we can't just handle everything in the handler implementations. No need to duplicate methods.

We can make all the handler methods virtual and override them just like the ones you've got here.

Reply:
So we're removing the check if the event is valid, then passing that on to the derived classes, correct?

Reply:
Currently (and desired): The handler contains the interactable check, then if valid, passes that on to the derived classes via the protected methods.

Reply:
Alright, I think that's okay, but the methods seem redundant (as you could still do the same things with a base call to check the validity). I'm interested in what others think as well.

Reply:
Yeah, I'm also not sure about "translating" the events here. I can understand having a "isInteracted" check before doing something but not sure this is the correct path.
Maybe if there was an "IMixedRealityInteractable" interface maybe...  But still not sure.

Reply:
Not really sure we need another interface here. It'd just be a copy of all the handers anyway

Reply:
FYI, added a diagram from HoloToolkit readme. The script which inherits InteractionReceiver can selectively implement event handlers and handle those types of events coming from multiple objects.